external_api/lib/dictionary_utils.py

```diff
@@ -25,45 +47,62 @@ def dictionary_update_languages(dictionary_name, new_language_list):
         raise APIHandlerException('No new languages provided. Nothing changed.')
 
     # fetch all words
-    word_list = get_dictionary_unique_words(dictionary_name)
-    if not word_list:
-        raise APIHandlerException('This dictionary does not have any records. Please verify the dictionary name')
+    values = get_entity_unique_values(entity_name=entity_name)
+    if not values:
+        raise APIHandlerException('This entity does not have any records. Please verify the entity name')
 
     records_to_create = []
     for language_script in languages_added:
         # create records for all words
-        for word in word_list:
+        for value in values:
             records_to_create.append({
-                'word': word,
+                'word': value,
                 'language_script': language_script,
                 'variants': []
             })
 
     datastore_obj = DataStore()
-    datastore_obj.add_data_elastic_search(dictionary_name, records_to_create)
+    datastore_obj.add_entity_data(entity_name, records_to_create)
 
     return True
 
 
-def get_dictionary_unique_words(
-    dictionary_name, empty_variants_only=False, word_search_term=None, variant_search_term=None
+def get_entity_unique_values(
+    entity_name, empty_variants_only=False, value_search_term=None, variant_search_term=None
 ):
     """
+    Get a list of unique values belonging to this entity
+    Args:
+        entity_name (str): Name of the entity for which unique values are to be fetched
+        empty_variants_only (bool): Flag to search for values with empty variants only
+        value_search_term (str): Search term to filter values from this entity data
+        variant_search_term (str): Search term to filter out variants from the entity data
+    Returns:
+        values_list (list): List of strings which are unique values in the entity
     """
     datastore_obj = DataStore()
-    return datastore_obj.get_dictionary_unique_words(
-        dictionary_name=dictionary_name,
-        word_search_term=word_search_term,
+    return datastore_obj.get_entity_unique_values(
+        entity_name=entity_name,
+        value_search_term=value_search_term,
         variant_search_term=variant_search_term,
         empty_variants_only=empty_variants_only
     )
 
 
-def get_records_from_word_list(dictionary_name, filtered_word_list=None):
+def get_records_from_values(entity_name, values=None):
+    """
+    Fetch entity data based for the specified values in that entity
+    Args:
+        entity_name (str): Name of the entity for which records are to be fetched
+        values (list): List of str values for which the data is to be fetched
+    Returns:
+        records (dict): dictionary with (key, value) as the value and language_code mapped
```

Comment:
Confusing statement. Possibly ill-formed
Also, this method returns a dict or list ? 

In any It is better to add a sample dict with keys for better understanding

Reply:
Added a sample return data